### PR TITLE
Add GetCurrentState() implementation for AWSConfig resource

### PR DESCRIPTION
**Pull request by tuommaki**
This is first step towards implementing `AWSConfig` resource in `cluster-operator`.

Towards https://github.com/giantswarm/giantswarm/issues/2148

### CODE REVIEW

File: service/controller/aws/v5/resource/awsconfig/common_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+var (
+	// Error to return when simulating unknown error returned from Kubernetes
+	// API client.
+	unknownAPIError = errors.New("Unknown error from k8s API")
+)
+
+func alwaysReturnErrorReactor(err error) k8stesting.Reactor {
+	return &k8stesting.SimpleReactor{
```
Comment (MarcelMue) [inline service/controller/aws/v5/resource/awsconfig/common_test.go]: This is pretty cool, I didn't know about this before :+1:

File: service/controller/aws/v5/resource/awsconfig/current.go [hunk 2]
```diff
@@ -2,11 +2,37 @@ package awsconfig
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (MarcelMue) [inline service/controller/aws/v5/resource/awsconfig/current.go]: Out of curiosity: Why did you choose `apismetav1`? I usually use `metav1` - I am wondering if there is a reason to use your version or if it's just preference.

Comment (tuommaki) [reply service/controller/aws/v5/resource/awsconfig/current.go]: No, could be that this originates from examples for testing reactor usage or then we have this somewhere lurking. I'm fine with both forms. :)

**Comment by tuommaki** [comment]
This is stale and won't happen with current designs. Closing.

**Pull request closed without merging**